dynamodb/README.md

```diff
@@ -0,0 +1,134 @@
+precision loss. For simplicity, we will first try to convert it into a NUMERIC
+type. If it fails, we convert it into a STRING type. 
+
+### `Null` Data Type
+
+In DynamoDB, an attribute can have a Null data type that represents an unknown
+or undefined state. Also, each row defines its own schema for attributes (not
+for primary keys). So attributes can be absent in rows. 
+
+We treat the above two cases the same as a Null value in Cloud Spanner. The
+cases that a column contains a Null value or a column is not present is an
+indication that this column should be nullable.
+
+### `List` and `Map`
+
+In Cloud Spanner, the most similar type to List and Map is
+[STRUCT](https://cloud.google.com/spanner/docs/data-types#struct_type), but it
+is not a valid column type (available for query but not for storage).
+Therefore, we encode them into a json string. 
+
+### Occasional Errors
+
+As no schema is forced when writing to a DynamoDB table, it can happen that a
+small number of data rows are incorrectly inserted. People do not realize that
+these data exist or their application can auto-fix these data. In Cloud Spanner,
+we have a strict schema for writes and we can’t write records with different
+data types. We define an error threshold - if the percentage of the data type is
+lower than or equal to an extremely low value (0.1%), then we consider the data
+type as an occasional error. It would not be considered as a candidate type for
+the column. In this case, we can filter a certain amount of noise when modeling
+the real schema. 
+
+### Multi-type Columns
+
+For a special scenario, we may get a column that has equal distribution of two
+data types. E.g., a column has 40% rows in String and 60% rows in Number. If we
+choose Number as its type, it may lead to 40% data loss in the data conversion.
+In the migration, we define a conflicting threshold on normalized rows (after
+removing Null data types and rows that the attribute is not present). By
+default, the conflicting threshold is 5% and if the percentages of two or more
+data types are greater than it, we would consider that the column has
+conflicting data types. As a safe choice, we define this column as a STRING type
+in Cloud Spanner. 
+
+## Data Conversion
+
+### A Scan for Entire Table
+
+A pass for the entire table is required. We will use Scan API to read data. Each
+read has a size limit up to 1MB. By using the returned token, we make a
+subsequent call to continue retrieving data from the table.
+
+The row result contains the data type and data itself. According to our
+[inferred schema](#schema-inference), we will parse the row to a format that
+Cloud Spanner can support. If the value parsing fails, we would drop the entire
+row and record it as bad data in the report. If a column does not appear or 
+column has a NULL data type, we would process this as a NULL value in Cloud Spanner. 
+The job is done when all rows are inserted into the table in Cloud Spanner. 
```

Comment:
Maybe drop this last sentence? I'm not sure it is needed.
Also, as it stands, it kind of suggests we only process one table.